sample-apps/delivery-customizations-sample-app/web/frontend/pages/rename/new.jsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from "react";
+import { useNavigate } from "@shopify/app-bridge-react";
+
+import { CustomizationForm, CustomizationPageLayout } from "../../components";
+import {
+  useCustomizationForm,
+  useCreateDeliveryCustomization,
+} from "../../hooks";
+
+export default function NewDeliveryCustomizationPage() {
+  const navigate = useNavigate();
+  const isNewCustomization = window.location.href.indexOf("new") > -1;
+
+  const { handleInputChange, setData, data: formData } = useCustomizationForm();
+
+  const { mutateAsync: createCustomization, isLoading } =
+    useCreateDeliveryCustomization();
+
+  const handleSubmit = async () => {
+    if (isLoading) return;
+    try {
+      await createCustomization({ payload: formData });
+      navigate("/");
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const primaryAction = {
+    disabled: isLoading,
+    onAction: handleSubmit,
+  };
+
+  useEffect(() => {
+    setData({
+      deliveryOptionName: "Express",
+      operation: "Rename",
+    });
+  }, []);
+
+  return (
+    <CustomizationPageLayout
+      title={`Rename delivery option`}
```

Comment:
Super nit: Since this is a static string, it should be using just a string literal

```suggestion
      title="Rename delivery option"
```